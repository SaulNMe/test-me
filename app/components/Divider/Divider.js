import React, { PureComponent } from 'react';
import { Text, View } from 'react-native';
import PropTypes from 'prop-types';
import { Colors  } from 'qualificame-native/app/styles';

import styles from './DividerStyle';

export default class Divider extends PureComponent {
  render() {
	return (
		<View style={[styles.divider, {backgroundColor: Colors[this.props.color]}]} />
	);
  }	
}

Divider.propTypes = {
	color: PropTypes.string
}

Divider.defaultProps = {
	color: 'gray'
}
