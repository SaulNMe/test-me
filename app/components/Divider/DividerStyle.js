import { StyleSheet } from 'react-native';
import { ApplicationStyles, Fonts, Colors, Metrics } from 'qualificame-native/app/styles';

export default StyleSheet.create({
	divider: {
		height: 1,
		color: Colors.light,
		width:'100%'
	}
});
