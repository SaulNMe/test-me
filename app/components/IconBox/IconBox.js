import React, { PureComponent } from 'react';
import { Text, View } from 'react-native';
import PropTypes from 'prop-types';
import { LinearGradient } from 'expo-linear-gradient';

import styles from './IconBoxStyle';
import { Feather } from '@expo/vector-icons';
import { Colors } from 'qualificame-native/app/styles';

export default class IconBox extends PureComponent {
	render() {
		let gradientColors;
		switch(this.props.bgColor) {
			case 'green':
				gradientColors = ['#0F9F92', '#12B184'];
			break;
			case 'blue':
				gradientColors = ['#154BBA', '#107EC2'];
			break;
			case 'orange':
				gradientColors = ['#E62808', '#FC680A'];
			break;
			case 'pink':
				gradientColors = ['#E31862', '#EB194D'];
			break;
			case 'gray':
				gradientColors = ['#ccc', '#ccc'];
			break;
			default:
				gradientColors = ['#fff', '#fff'];
			break;
		}
		return (
			<LinearGradient colors={gradientColors} style={[styles.defaultSize, styles.centerObjects]} start={[0, 0]} end={[1, 0]}>
				<Feather name={this.props.iconName} size={24} color={Colors.white}/>
			</LinearGradient>
		);
	}
}

	IconBox.propTypes = {
		bgColor: PropTypes.string,
		iconName: PropTypes.string

	}

	IconBox.defaultProps = {
		bgColor: 'green',
		iconName: 'x'
	}
