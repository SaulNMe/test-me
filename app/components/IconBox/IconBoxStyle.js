import { StyleSheet } from 'react-native';
import { ApplicationStyles, Fonts, Colors, Metrics } from 'qualificame-native/app/styles';

export default StyleSheet.create({
	...ApplicationStyles.flexBox,
	defaultSize:{
		width: 32,
		height: 32,
		borderRadius: 5
	}
});
