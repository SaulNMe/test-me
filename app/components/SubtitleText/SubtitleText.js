import React, { PureComponent } from 'react';
import { Text } from 'react-native';
import PropTypes from 'prop-types';
import { Fonts, Colors } from 'qualificame-native/app/styles';
import styles from './SubtitleTextStyle';

export default class SubtitleText extends PureComponent {
	render() {
		const textColor = { color: Colors[this.props.color] };
		const weight = { fontWeight: Fonts.weight[this.props.weight] };
		const align = { textAlign: this.props.align };
		return (
			<React.Fragment>
				<Text style={[styles.text, textColor, weight, align]}>{this.props.text}</Text>
			</React.Fragment>
		);
	}
}

	SubtitleText.propTypes = {
		text: PropTypes.string,
		color: PropTypes.string,
		weight: PropTypes.string,
		align: PropTypes.string,
	}

	SubtitleText.defaultProps = {
		text: "",
		color: "dark",
		weight: "regular",
		align: "left"
	}
