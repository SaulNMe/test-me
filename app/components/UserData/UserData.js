import React, { PureComponent } from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';

import styles from './UserDataStyle';
import IconBox from 'qualificame-native/app/components/IconBox';
import Divider from 'qualificame-native/app/components/Divider';
import SubtitleText from 'qualificame-native/app/components/SubtitleText';


export default class UserData extends PureComponent {
	render() {
		return (
			<View style={styles.bgColor}>
				<Divider />
				<View style={[styles.row, styles.baseHorizontalMargin, styles.alignItemsCenter]}>
					<IconBox 
						bgColor="green"
						iconName="user"
					/>
					<View style={[styles.baseLeftMargin, styles.smallVerticalMargin]}>
						<SubtitleText 
							text={this.props.name}
							color="dark"
						/>
						<SubtitleText 
							text={this.props.email}
							color="lightBlack"
							weight="light"
						/>
					</View>
				</View>
					<Divider />
			</View>
		);
	}
}

	UserData.propTypes = {
		name: PropTypes.string,
		email: PropTypes.string
	}

	UserData.defaultProps = {
		name: "",
		email: ""
	}
