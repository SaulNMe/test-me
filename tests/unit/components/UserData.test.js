import React from 'react';
import { shallow } from 'enzyme';

import UserData from 'qualificame-native/app/components/UserData';
import SubtitleText from 'qualificame-native/app/components/SubtitleText';
import Divider from 'qualificame-native/app/components/Divider';
import IconBox from 'qualificame-native/app/components/IconBox';
import { Text } from 'react-native';
import { Feather } from '@expo/vector-icons';


function createTestProps (props) {
	return {
		// common props
		name: 'name',
		email: 'email',
		// allow to override common props
		...props,
	}
}

describe('rendering', () => {
	let wrapper;
	const createWrapper = props => shallow(<UserData {...props}/>);
	beforeEach(() => {
		const props = createTestProps()
		wrapper = createWrapper(props);
	});

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('displays name ', () => {
		expect( wrapper.find(SubtitleText).first().props().text ).toEqual('name');
	});

	it('displays email', () => {
		expect( wrapper.find(SubtitleText).last().props().text ).toEqual('email');
	});

	describe('when props are not set', () => {
		beforeEach(() => {
			wrapper = createWrapper();
		});
		
		it('displays no text', () => {
			expect(wrapper.find(SubtitleText).first().prop('text')).toBeFalsy();
			expect(wrapper.find(SubtitleText).last().prop('text')).toBeFalsy();
		});
		
		it('displays iconBox', () => {
			expect(wrapper.find(IconBox).exists()).toBeTruthy();
		});		

		it('displays both dividers', () => {
			expect(wrapper.find(Divider).getElements().length).toEqual(2);
		});
	});
});

