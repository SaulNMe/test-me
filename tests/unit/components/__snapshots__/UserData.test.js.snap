// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<Component>
  <Divider
    color="gray"
  />
  <Component
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "marginHorizontal": 16,
        },
        Object {
          "alignItems": "center",
        },
      ]
    }
  >
    <IconBox
      bgColor="green"
      iconName="user"
    />
    <Component
      style={
        Array [
          Object {
            "marginLeft": 16,
          },
          Object {
            "marginVertical": 8,
          },
        ]
      }
    >
      <SubtitleText
        align=""
        color="dark"
        text="name"
        weight=""
      />
      <SubtitleText
        align=""
        color="dark"
        text="email"
        weight="light"
      />
    </Component>
  </Component>
  <Component
    style={
      Object {
        "marginBottom": 16,
      }
    }
  >
    <Divider
      color="gray"
    />
  </Component>
</Component>
`;
